packages/@lwc/ssr-compiler/src/compile-template/shared.ts

```diff
@@ -153,20 +154,27 @@ export function getChildAttrsOrProps(
                     // `spellcheck` string values are specially handled to massage them into booleans:
                     // https://github.com/salesforce/lwc/blob/574ffbd/packages/%40lwc/template-compiler/src/codegen/index.ts#L445-L448
                     literalValue = literalValue.toLowerCase() !== 'false';
+                } else if (name.toLowerCase() === 'tabindex') {
+                    // Global HTML "tabindex" attribute is specially massaged into a stringified number
+                    // This follows the historical behavior in api.ts:
+                    // https://github.com/salesforce/lwc/blob/f34a347/packages/%40lwc/engine-core/src/framework/api.ts#L193-L211
+                    const shouldNormalize = parseInt(literalValue) > 0;
+                    literalValue = shouldNormalize ? '0' : literalValue;
                 }
                 return b.property('init', key, b.literal(literalValue));
             } else if (value.type === 'Literal' && typeof value.value === 'boolean') {
                 if (name === 'class') {
                     return; // do not render empty `class=""`
                 }
-
                 return b.property('init', key, b.literal(type === 'Attribute' ? '' : value.value));
             } else if (value.type === 'Identifier' || value.type === 'MemberExpression') {
                 let propValue = expressionIrToEs(value, cxt);
-
                 if (name === 'class') {
                     cxt.import('normalizeClass');
                     propValue = b.callExpression(b.identifier('normalizeClass'), [propValue]);
+                } else if (name.toLowerCase() === 'tabindex') {
```

Comment:
nit: there is more than one `name.toLowerCase()` showing up in this function; probably could define `nameLower` variable.